src/main/java/opinion/evaluators/SentenceEvaluator.java

```diff
@@ -0,0 +1,24 @@
+package opinion.evaluators;
+
+import opinion.sentences.Sentence;
+
+public abstract class SentenceEvaluator {
+
+    /**
+     *
+     * @return the result of applying this evaluator to a sentence
+     */
+    EvaluatorResult applyEvaluator(Sentence sentence) {
+        return new EvaluatorResult.Builder()
+                .setEvaluatorApplies(evaluatorApplies(sentence))
+                .setEvaluatorDescription(getEvaluatorExplanation())
+                .setEvaluatorType(getEvaluatorType())
+                .build();
+    }
+
+    protected abstract String getEvaluatorExplanation();
+
+    public abstract EvaluatorResult.EvaluatorType getEvaluatorType();
+
+    public abstract boolean evaluatorApplies(Sentence sentence);
```

Comment:
The motivation behind this refactor was realizing that many solutions to classifying sentences don't approach the problem from

```
if (ruleApplies(sentence)) {
    // we can make a conclusion i.e. opinion or fact
}
```

but instead:

`classification = evaluateSentence(sentence)`

That is to say, there is no question whether an evaluator applies. An evaluator always returns something, and specifically can retutrn either `opinion` or `fact` (though we should also support `no result`). So while the renaming here is good, we also need to refactor method interfaces a bit.